draft-irtf-cfrg-bbs-signatures.md

```diff
@@ -898,25 +893,28 @@ Inputs:
 
 Outputs:
 
-- challenge, a scalar or INVALID.
+- challenge, a scalar.
+
+Deserialization:
+
+1. R = length(i_array)
+2. (i1, ..., iR) = i_array
+3. (msg_i1, ..., msg_iR) = msg_array
+
+ABORT if:
+
+1. R > 2^64 - 1 or R != length(msg_array)
+2. length(ph) > 2^64 - 1
+3. for i in i_array, i > 2^64 - 1
```

Comment:
I believe this would be checked by `serialize`?

Reply:
Yes. Iv added these checks here to ABORT instead of returning INVALID (which we would have to do if we were relying on `serialize` for the checks).

That way we don't have to check the result of `calculate_challenge` and `calculate_domain` in the core operations.